src/gui/main.cpp

```diff
@@ -60,7 +61,18 @@ int main(int argc, char* argv[])
 
     qDebug() << "Qt message ";
 
+    CanDeviceManager m;
     MainWindow w;
+
+    QObject::connect(&w, SIGNAL(requestAvailableDevices(QString)), &m, SLOT(fetchAvailableDevices(QString)));
```

Comment:
Please don't use SIGNAL, SLOT macros, use &CanDeviceManager::selectCanDevice. 
This catches errors during compile time, not runtime